### PR TITLE
Fuse search

### LINKED ISSUE #23: Searching Functionality
### Goal: Use Fuzzy Searching to implement a search bar feature that filters through psychiatrists.

Work on branch `fuse-search`. We will be using Fuse.js to implement fuzzy matching. I have implemented a basic search bar component in `SearchBar.tsx`, and a Psychiatrist List component in `PsychiatristList.tsx`. Based on the query parameters in frontend/src/pages/pyschiatrists/[query].tsx, which is a dynamic route, you have to render a list of psychiatrists according to updated Mid-fis.

Familiarize with some key concepts:

- What is useMemo? https://react.dev/reference/react/useMemo#reference
- Dynamic Routes: https://nextjs.org/docs/routing/dynamic-routes
- Fuse.js documentation: https://fusejs.io/api/options.html

We want to implement this with Firebase.

- [x] Move the ticket
- [x] Optimize searching functionality based on different keys and options. Try out extended search techniques.
- [x] Finish rendering the psychiatrist list properly based on query parameters.
- [x] Testing and replication
- [x] Make PR

### PR DESCRIPTION
**Pull request by jessicajcho**
## Overview
resolves #23 
Implemented fuzzy searching and created searchbar component & filter dropdown components

## Changes Made

- created psychiatrist cards with buttons to save psychiatrist / message psychiatrist (buttons have no functionality)
- created search bar that searches by name or title of psychiatrist. renders result upon pressing 'enter'
- added filter dropdowns
- 
## Test Coverage
Manual testing.

## Screenshots (delete if not applicable)

<details>

  <summary>Listing Page</summary>

 image

</details>

<details>

  <summary>Filter Drop-downs</summary>

image

</details>

<details>

  <summary>Search Results</summary>

image

</details>

### CODE REVIEW

**Review by p-m-ishra — COMMENTED** [review]
PR looks good and organized. Feature is working. Great work!

**Comment by sriramanved** [comment]
This branch did not receive a review from @hahnandrew. @p-m-ishra, please provide more of a thorough review in the future. @jessicajcho @swinshu, the comments left on your PR should be reviewed and addressed within a reasonable time frame. I think we can improve on communication here, but nonetheless good job finishing this ticket.

**Comment by sriramanved** [comment]
This branch did not receive a review from @hahnandrew. @p-m-ishra, please provide more of a thorough review in the future. @jessicajcho @swinshu, the comments left on your PR should be reviewed and addressed within a reasonable time frame. I think we can improve on communication here, but nonetheless good job finishing this ticket.

File: frontend/src/components/PsychiatristList.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  specialty: string[];
+  description: string
+}
+
+
+interface PsychiatristListProps {
+  results: Psychiatrist[];
+}
+
+const PsychiatristList: React.FC<PsychiatristListProps> = ({ results }) => {
```
Comment (hahnandrew) [inline frontend/src/components/PsychiatristList.tsx:30]: love the code comments and how they make this huge div more readable. might be even better to add spacing to segment the code even more for readability for the future devs working on this file

File: frontend/src/components/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                value={searchTerm}
+                // updates the search bar's text as the user types into it
+                onChange={handleChange} className="input input-bordered" />
+              <button className="btn btn-square">
+                <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor"><path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z" /></svg>
+              </button>
+            </div>
+          </div>
+        </div>
+        {/* filter dropdowns for availability */}
```
Comment (hahnandrew) [inline frontend/src/components/SearchBar.tsx:36]: great code documentation, very readable

**Pull request merged**